File: 3498-reverse-degree-of-a-string/3498-reverse-degree-of-a-string.js
function reverseDegree(s) {
    let totalScore = 0;

    for (let position = 1; position <= s.length; position++) {
        const currentChar = s.charAt(position - 1);

        const reverseAlphabeticalValue =
            26 - (currentChar.charCodeAt(0) - 'a'.charCodeAt(0));

        totalScore += position * reverseAlphabeticalValue;
    }

    return totalScore;
}

File: 3498-reverse-degree-of-a-string/3498-reverse-degree-of-a-string.java
class Solution {
    public int reverseDegree(String s) {
        int stringLength = s.length();
        int totalReverseDegree = 0;
      
        // Iterate through each character position (1-based indexing)
        for (int position = 1; position <= stringLength; position++) {
            // Get the character at current position (convert to 0-based index)
            char currentChar = s.charAt(position - 1);
          
            // Calculate reverse alphabetical value (distance from 'z')
            // 'a' = 26, 'b' = 25, ..., 'z' = 1
            int reverseAlphabeticalValue = 26 - (currentChar - 'a');
          
            // Add weighted value (position * reverse value) to total
            totalReverseDegree += position * reverseAlphabeticalValue;
        }
      
        return totalReverseDegree;
    }
}
